utils/s3/credentials_providers/instance_profile_credentials_provider.cc

```diff
@@ -34,7 +34,7 @@ future<> instance_profile_credentials_provider::reload() {
 
 future<> instance_profile_credentials_provider::update_credentials() {
     auto factory = std::make_unique<utils::http::dns_connection_factory>(ec2_metadata_ip, port, false, ec2_md_logger);
-    http::experimental::client http_client(std::move(factory), 1, http::experimental::client::retry_requests::yes);
+    retryable_http_client http_client(std::move(factory), 1, {}, http::experimental::client::retry_requests::yes, retry_strategy);
```

Comment:
I meant [this](https://godbolt.org/z/cnKK7rMYb).

Reply:
yes, this is what I mean. The fact that `std::function` has implicit constructor from `nullptr_t`  does not mean one has to rush and use it and sacrifice the readability readability and elegance of using default constructed std::function without a callable in it.
Bottom line, what do you achive by choosing `function(std::nullptr_t)noexcept;` over `function() noexcept;`

Reply:
For _me_ passing `nullptr` express better the intention: there is no intention to use this parameter, regardless of the type of the parameter. I.e when you see a `nullptr` this is an indication that this parameter is not going to be used. On the other hand an empty object means that the default constructed object _will be_ in use.
Yes, in this specific case it means the same.
Overall it is just a perception, nothing with the "real" code. I guess that for `std::function`, the empty object and the `nullptr_t`  versions are equivalent in all ways.